tck/src/main/java/org/microprofile/concurrency/tck/TckTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.microprofile.concurrency.tck;
+
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.concurrent.ThreadContext;
+import org.eclipse.microprofile.concurrent.spi.ConcurrencyProvider;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+public class TckTest {
+
+    @Deployment
+    public static WebArchive createDeployment() {
```

Comment:
Any specific reason why the deployment is `WebArchive`?
Because as it is written now, this will result in duplicate `TckTest` class in the archive. E.g. the structure is as follows:

someTestArchive.war
+-- _WEB-INF
|   +-- classes/org.micro..../TckTest.class
|   +-- lib/createdJarArchive.jar
|         +-- META-INF/beans.xml
|         +-- org.micro..../TckTest.class

I am not an Arq. expert, but I think the reason is that Arquillian automatically adds the test class into the root archive.
The only exception is when it is run in a [client mode](https://docs.jboss.org/author/display/ARQ/Test+run+modes).

Reply:
Frankly I don't know, I can try to change it and see if it still works. I copied this set up from mp-config.

Reply:
Well, you can either go with just jar, where the test class will also be automatically added or make it run in client mode.
From the TCK runner in smallrye I saw that you are using weld-embedded anyway.
That's basically starting Weld SE container and assumes "se-like structure" which means flat one (at least I think it does :)). If that's the case, having WAR has no effect versus JAR.

Reply:
On second thought, WAR is probably what users use more often.
So we might just fix the duplicate class issue and leave it at that.